src/databricks/labs/ucx/source_code/files.py

```diff
@@ -44,12 +48,79 @@ def __repr__(self):
         return f"<LocalFile {self._path}>"
 
 
+class LocalDirectory(SourceContainer):
+    def __init__(self, path: Path, file_loader: FileLoader, dir_loader: DirectoryLoader):
+        self._path = path
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+
+    @property
+    def path(self) -> Path:
+        return self._path
+
+    def build_dependency_graph(self, parent: DependencyGraph) -> list[DependencyProblem]:
+        # don't directly scan non-source directories, let it be done for relevant imports only
+        if self._path.name in {"__pycache__", ".git", ".github", ".venv", "site-packages"}:
+            return []
+        return list(self._build_dependency_graph(parent))
+
+    def _build_dependency_graph(self, parent: DependencyGraph) -> Iterable[DependencyProblem]:
+        for child_path in self._path.iterdir():
+            loader = self._dir_loader if child_path.is_dir() else self._file_loader
+            dependency = Dependency(loader, child_path)
+            yield from parent.register_dependency(dependency).problems
+
+    def __repr__(self):
+        return f"<LocalDirectory {self._path}>"
+
+
+class LocalFilesLinter:
+
+    def __init__(
+        self,
+        file_loader: FileLoader,
+        dir_loader: DirectoryLoader,
+        path_lookup: PathLookup,
+        dependency_resolver: DependencyResolver,
+        languages_factory: Callable[[], Languages],
+    ) -> None:
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+        self._path_lookup = path_lookup
+        self._dependency_resolver = dependency_resolver
+        self._extensions = {".py": Language.PYTHON, ".sql": Language.SQL}
+        self._languages_factory = languages_factory
+
+    def lint(self, path: Path) -> list[LocatedAdvice]:
+        return list(self._lint(path))
+
+    def _lint(self, path: Path) -> Iterable[LocatedAdvice]:
+        loader = self._dir_loader if path.is_dir() else self._file_loader
+        dependency = Dependency(loader, path)
+        graph = DependencyGraph(dependency, None, self._dependency_resolver, self._path_lookup)
+        container = dependency.load(self._path_lookup)
+        assert container is not None  # because we just created it
+        problems = container.build_dependency_graph(graph)
+        for problem in problems:
+            path = Path('UNKNOWN') if problem.is_path_missing() else problem.source_path.absolute()
+            yield LocatedAdvice(path=path, advice=problem.as_advisory())
```

Comment:
```suggestion
            problem_path = path if problem.is_path_missing() else problem.source_path.absolute()
            yield LocatedAdvice(problem.as_advisory(), problem_path)
```
bug: otherwise method argument is shadowed and overwritten.

Reply:
not a bug yet but well spotted